x/adminmodule/keeper/proposal_legacy.go

```diff
@@ -29,16 +28,36 @@ func (k Keeper) SubmitProposalLegacy(ctx sdk.Context, content govv1beta1types.Co
 	}
 
 	headerTime := ctx.BlockHeader().Time
-
 	// submitTime and depositEndTime would not be used
 	proposal, err := govv1beta1types.NewProposal(content, proposalID, headerTime, headerTime)
 	if err != nil {
-		return govv1beta1types.Proposal{}, err
+		return govv1beta1types.Proposal{}, sdkerrors.Wrap(err, "failed to create proposal struct")
 	}
 
+	handler := k.RouterLegacy().GetRoute(proposal.ProposalRoute())
+	cacheCtx, writeCache := ctx.CacheContext()
+
+	// The proposal handler may execute state mutating logic depending
+	// on the proposal content. If the handler fails, no state mutation
+	// is written and the error message is returned.
+	err = handler(cacheCtx, content)
+	if err == nil {
+		proposal.Status = govv1beta1types.StatusPassed
+
+		// The cached context is created with a new EventManager. However, since
+		// the proposal handler execution was successful, we want to track/keep
+		// any events emitted, so we re-emit to "merge" the events into the
+		// original Context's EventManager.
+		ctx.EventManager().EmitEvents(cacheCtx.EventManager().Events())
```

Comment:
this seems not to be necessary because this is handled in the writeCache func:
<img width="653" alt="image" src="https://github.com/neutron-org/admin-module/assets/34917380/59453885-9289-4dc8-8ff2-a406d979fcb6">